pycoast/cw_base.py

```diff
@@ -862,38 +864,47 @@ def add_overlay_from_dict(self, overlays, area_def, cache_epoch=None, background
-            font = overlays['grid'].get('font', None)
-            font_size = int(overlays['grid'].get('font_size', 10))
-            grid_kwargs = {}
-            if is_agg:
-                width = float(overlays['grid'].get('width', 1.0))
-                grid_kwargs["width"] = width
-
+            if 'major_lonlat' in overlays['grid'] or 'minor_lonlat' in overlays['grid']:
+                Dlonlat = overlays['grid'].get('major_lonlat', (10.0, 10.0))
+                dlonlat = overlays['grid'].get('minor_lonlat', (2.0, 2.0))
+            else:
+                Dlonlat = (overlays['grid'].get('Dlon', 10.0), overlays['grid'].get('Dlat', 10.0))
+                dlonlat = (overlays['grid'].get('dlon', 2.0), overlays['grid'].get('dlat', 2.0))
+            outline = overlays['grid'].get('outline', 'white')
             write_text = overlays['grid'].get('write_text', True)
             if isinstance(write_text, str):
                 write_text = write_text.lower() in ['true', 'yes', '1', 'on']
-            outline = overlays['grid'].get('outline', 'white')
+            # This should close PyCoast issue #52
+            font = overlays['grid'].get('font', None)
+            font_size = int(overlays['grid'].get('font_size', 10))
+            fill = overlays['grid'].get('fill', 'white')
+            fill_opacity = overlays['grid'].get('fill_opacity', 255)
             if isinstance(font, str):
                 if is_agg:
                     from aggdraw import Font
-                    font = Font(outline, font, size=font_size)
+                    font = Font(fill, font, opacity=fill_opacity, size=font_size)
                 else:
                     from PIL.ImageFont import truetype
                     font = truetype(font, font_size)
-            fill = overlays['grid'].get('fill', None)
             minor_outline = overlays['grid'].get('minor_outline', 'white')
             minor_is_tick = overlays['grid'].get('minor_is_tick', True)
             if isinstance(minor_is_tick, str):
                 minor_is_tick = minor_is_tick.lower() in ['true', 'yes', '1']
             lon_placement = overlays['grid'].get('lon_placement', 'tb')
             lat_placement = overlays['grid'].get('lat_placement', 'lr')
 
-            self.add_grid(foreground, area_def, (lon_major, lat_major),
-                          (lon_minor, lat_minor),
+            grid_kwargs = {}
+            if is_agg:
+                width = float(overlays['grid'].get('width', 1.0))
+                minor_width = float(overlays['grid'].get('minor_width', 0.5))
+                outline_opacity = overlays['grid'].get('outline_opacity', 255)
+                minor_outline_opacity = overlays['grid'].get('minor_outline_opacity', 255)
+                grid_kwargs['width'] = width
+                grid_kwargs['minor_width'] = minor_width
+                grid_kwargs['outline_opacity'] = outline_opacity
+                grid_kwargs['minor_outline_opacity'] = minor_outline_opacity
```

Comment:
So these are agg-specific kwargs that aren't available in the PIL writer?

Reply:
No, PIL is very limited, width is always 1 and no opacity.

Reply:
You mean "yes"? PIL is very limited so these options aren't available in PIL, correct?